swap_meet/vendor.py

```diff
@@ -1,2 +1,63 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        self.inventory.remove(item)
+        return item
+
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        self.add(their_item)
+        self.remove(my_item)
+        other_vendor.add(my_item)
+        other_vendor.remove(their_item)
+        return True
+
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False
+        their_first_item = other_vendor.inventory[0]
+        my_first_item = self.inventory[0]
+        self.swap_items(other_vendor, my_first_item, their_first_item)
+        return True
+
+    def get_by_category(self, category):
+        objects_of_category = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                objects_of_category.append(item)
+        return objects_of_category
+
+    def get_best_by_category(self, category):
+        best_item = None
+        best_item_condition = 0
+        for item in self.inventory:
+            if item.get_category() == category:
+                if item.condition > best_item_condition:
+                    best_item = item
+                    best_item_condition = item.condition
+        return best_item
```

Comment:
Once you're comfortable with the version that uses loops, you can also write this with `max` and rely on your `get_by_category` function:

```python
by_category = self.get_by_category(category)
return max(self.inventory, key=lambda item: item.condition) if by_category else None
```